### PR TITLE
adds merge_pop_numbers function

### PR DESCRIPTION
**Pull request by joshzarrabi**
* merges `population` and `population_pct` columns into an existing df
* for #1474 issues

## Motivation and Context

So we can have a consistent function to easily merge in population numbers.

## How has this been tested?

Yes

## Types of changes

- New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: python/ingestion/dataset_utils.py [hunk 1]
```diff
@@ -124,3 +124,30 @@ def merge_fips_codes(df):
     df = df.rename(columns={'state_fips_code': std_col.STATE_FIPS_COL}).reset_index(drop=True)
 
     return df
+
+
+def merge_pop_numbers(df, demo, loc):
+    """Merges the corresponding `population` and `population_pct` column into the given df
+
+      df: a pandas with demogrpahic and a `state_fips` column
```
Comment (benhammondmusic) [inline python/ingestion/dataset_utils.py:132]: typo

Comment (benhammondmusic) [reply python/ingestion/dataset_utils.py:132]: We should probably specify that the df is expected to have the items in the `demo` column be a race code (like `WHITE_NH`)

**Review by benhammondmusic — APPROVED** [review]
super helpful going forward. thanks

File: python/ingestion/dataset_utils.py (hunk 1 above)
Comment (benhammondmusic) [reply python/ingestion/dataset_utils.py]: Sure. But I mean we should specify the exact format expected for that column's data. For race it needs to be race code, for sex it needs to be a string `Male` or `Female`, for age is has to be `10-14` or whatever. Like where we someone using that fn know to look for what was expected as a value in that col

Comment (joshzarrabi) [reply python/ingestion/dataset_utils.py]: I see. Maybe we should make like a demographic enum or something to enforce it. I think its a bit of a bigger issue than can be solved in the docstring of this function TBH

**Pull request merged**